File: test90_out/local.mcu/GlobalInterrupts/GlobalInterrupts.c

/*
 * ======== string literals ========
 */

static const char local_mcu_GlobalInterrupts__s__fileName[] = "GlobalInterrupts.p";
static const char local_mcu_GlobalInterrupts__s__1[] = "GlobalInterrupts not restored\n";
static const char local_mcu_GlobalInterrupts__s__2[] = "GlobalInterrupts";
static const char local_mcu_GlobalInterrupts__s__0[] = "GlobalInterrupts restored\n";

/*
 * ======== function definitions ========
 */

void local_mcu_GlobalInterrupts_restore__E( bool state ) {
    if ((state)) {
        test90_Test_03_pollen__printStr((string)local_mcu_GlobalInterrupts__s__0);
    }
    else {
        test90_Test_03_pollen__printStr((string)local_mcu_GlobalInterrupts__s__1);
    }
}

void local_mcu_GlobalInterrupts_enable__E() {
    (local_mcu_GlobalInterrupts_enabled__V) = true;
}

void local_mcu_GlobalInterrupts_disable__E() {
    (local_mcu_GlobalInterrupts_enabled__V) = false;
}

bool local_mcu_GlobalInterrupts_isEnabled__E() {
    return(  (local_mcu_GlobalInterrupts_enabled__V)  );
}

void local_mcu_GlobalInterrupts_targetInit__I() {
}

File: test90_out/local.mcu/GlobalInterrupts/GlobalInterrupts.js

/*
 * ======== MODULE local.mcu.GlobalInterrupts ========
 */

var $$c = function() {
    const GlobalInterrupts = {};
    GlobalInterrupts.GlobalInterrupts = GlobalInterrupts
    GlobalInterrupts.$name = 'local.mcu.GlobalInterrupts';
    GlobalInterrupts.pollen$used = 0;
    GlobalInterrupts.enable = new $$Ref('local_mcu_GlobalInterrupts_enable__E');
    GlobalInterrupts.disable = new $$Ref('local_mcu_GlobalInterrupts_disable__E');
    GlobalInterrupts.restore = new $$Ref('local_mcu_GlobalInterrupts_restore__E');
    GlobalInterrupts.isEnabled = new $$Ref('local_mcu_GlobalInterrupts_isEnabled__E');
    GlobalInterrupts.targetInit = new $$Ref('local_mcu_GlobalInterrupts_targetInit__I');
    GlobalInterrupts.$$hostInit = function() {
        var $$text = '';
        return $$text;
    }
    GlobalInterrupts.pollen__uses__ = function() {
    }
    GlobalInterrupts.$$privateInit = function() {
        GlobalInterrupts.enabled = false;
    }
    return GlobalInterrupts;
}
var $$u = $$c();
$units['local.mcu.GlobalInterrupts'] = $$u;
$units.push($$u);

